### PR TITLE
[snapshots] use user locale in timestamp output

### PR DESCRIPTION
**Pull request by sharder996**

### CODE REVIEW

File: src/client/cli/formatter/format_utils.cpp [hunk 1]
```diff
@@ -110,3 +110,19 @@ void mp::format::filter_aliases(google::protobuf::RepeatedPtrField<multipass::Fi
...
+{
+}
+
+std::string mp::FormatUtils::convert_to_localtime(const google::protobuf::Timestamp& timestamp) const
+{
+    std::ostringstream oss;
+    std::time_t t = google::protobuf::util::TimeUtil::TimestampToTimeT(timestamp);
+
+    // ISO 8601: YYYY-MM-DDThh:mm:ss±hhmm
+    oss << std::put_time(std::localtime(&t), "%FT%T%z");
```
Comment (sharder996) [inline src/client/cli/formatter/format_utils.cpp:125]: Currently have this set up to print out the UTC offset. Excluding `±hhmm` looks a bit more presentable, but then means that the datetime is ambiguous on whether it defines UTC time or localtime. See [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Local_time_(unqualified)).

**Comment by ricab** [comment]
Thanks for this @sharder996. I will enqueue this with a little lower priority to focus on PRs that were already on flight. But hopefully we can still get it into the MVP.

**Comment by ricab** [comment]
Hey @sharder, could you please rebase the relevant commits here? Thank you!

**Review by ricab — COMMENTED** [review]
Hey @sharder996, thanks for this PR! It looks good overall, both structurally and in detail. However, you are using local time, but not a locale yet. The locale is important in this instance mainly to avoid ambiguity in the date (is 2022-11-02 the same as Nov 2nd or Feb 11th?); to a lesser extent, in the time too (is 1:15 am or pm?). 

In any case, this is certainly valuable as is. If it turns out to be difficult to use the locale, we can skip that and accept. But please have a look at my inline comment.

**Review by ricab — APPROVED** [review]
Thanks @sharder996! This doesn't work within a snap, but I think we can get it in and deal with that separately. I have only a tiny request inline, then we can merge.

**Comment by townsend2010** [comment]
bors ping

**Comment by townsend2010** [comment]
Ok, CI is all green as are [integration tests](https://github.com/canonical/multipass/actions/runs/6885616587).  Merging...

**Pull request merged**